triggers/runner/src/main/scala/com/daml/trigger/Runner.scala

```diff
@@ -240,26 +251,56 @@ object Converter {
     }
   }
 
-  private def toTemplateId(triggerIds: TriggerIds, v: SValue): Either[String, Identifier] = {
+  private def toIdentifier(v: SValue): Either[String, Identifier] = {
+    v match {
+      case SRecord(_, _, vals) => {
+        assert(vals.size == 3)
+        for {
+          packageId <- toText(vals.get(0)).flatMap(PackageId.fromString)
+          moduleName <- toText(vals.get(1)).flatMap(DottedName.fromString)
+          entityName <- toText(vals.get(2)).flatMap(DottedName.fromString)
+        } yield Identifier(packageId, QualifiedName(moduleName, entityName))
+      }
+      case _ => Left(s"Expected Identifier but got $v")
+    }
+  }
+
+  private def toTemplateId(v: SValue): Either[String, Identifier] = {
```

Comment:
This function is somewhat different from the other `toX` functions in that it doesn’t convert the actual value but only the metadata stored in the record. Maybe `extractTemplateId` would be a better name?

Reply:
Agreed, that's a better name.